lib/chopperbot/message_builder.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Chopperbot.MessageBuilder do
+  alias Chopperbot.Split
+
+  @callback validate_text_input(String.t()) :: {:ok, String.t()} | {:error, String.t()}
+  @callback build_ok_message([Split.Order.t()]) :: map()
+  @callback build_error_message(String.t()) :: map()
+
+  @doc """
+  Build a message from the given text and platform.
+  """
+  @spec build(String.t(), for: atom()) :: map()
+  def build(text, for: platform) do
+    message_builder = get_message_builder_from_platform(platform)
+
+    with {:ok, validated_text} <- message_builder.validate_text_input(text),
+         {:ok, orders} <- Split.run(validated_text) do
```

Comment:
Is it feel a little strange for the actual calculation to be in the message builder? It might feel more intuitive for me for the message builder to accept the specific format of data (ex. list of calculated orders), then return the formatted text.

Reply:
IMO, the `Split` is the 1 main module. Which can have `Remind`, `Weather` or some other utilities module later? 

So is it better to structure with the Split on top, and put the message builder inside the `Split` module like `Chopperbot.Split.LineMessageBuilder`?

Reply:
Nice idea krub 👍 
I feel bad about presenting `Order.t` outside `Split` folder as well 😅 
Thx for the suggestion ✌️